packages/web/src/common/store/smart-collection/sagas.ts

```diff
@@ -34,22 +35,36 @@ const { getUserId } = accountSelectors
 const COLLECTIONS_LIMIT = 25
 
 function* fetchHeavyRotation() {
-  const explore = yield* getContext('explore')
-  const topListens = yield* call([explore, 'getTopUserListens'])
+  const currentUserId = yield* select(getUserId)
+  const apiClient = yield* getContext('apiClient')
+  const userListeningHistoryMostListenedByUser = yield* call(
+    [apiClient, apiClient.getUserTrackHistory],
+    {
+      userId: currentUserId!, // userId
+      currentUserId,
+      limit: 20, // limit
+      offset: 0, // offset
+      sortMethod: 'most_listens_by_user'
+    }
+  )
+
+  const mostListenedTracks = userListeningHistoryMostListenedByUser
+    .map((listeningHistoryWithTrack) => listeningHistoryWithTrack.track)
+    .filter(removeNullable)
 
   const users = yield* call(
     retrieveUsers,
-    topListens.map((t) => t.userId)
+    mostListenedTracks.map((t) => t.owner_id)
   )
 
-  const trackIds = topListens
+  const trackIds = mostListenedTracks
     .filter(
       (track) =>
-        users.entries[track.userId] &&
-        !users.entries[track.userId].is_deactivated
+        users.entries[track.owner_id] &&
+        !users.entries[track.owner_id].is_deactivated
     )
-    .map((listen) => ({
-      track: listen.trackId
+    .map((track) => ({
```

Comment:
do we need to filter tracks by deleted/unavailable etc?

Reply:
hmm it didn't filter them out before but thats probably a good call, will add